pkg/collector/corechecks/cluster/orchestrator/orchestrator.go

```diff
@@ -340,21 +345,45 @@ func (o *OrchestratorCheck) processNodes(sender aggregator.Sender) {
 	}
 	groupID := atomic.AddInt32(&o.groupID, 1)
 
-	messages, err := processNodesList(nodesList, groupID, o.orchestratorConfig, o.clusterID)
+	nodesMessages, clusterMessage, err := processNodesList(nodesList, groupID, o.orchestratorConfig, o.clusterID, o.apiClient)
 	if err != nil {
 		o.Warnf("Unable to process node list: %s", err) //nolint:errcheck
 		return
 	}
 
+	sendNodesMetadata(sender, nodesList, nodesMessages, o.clusterID)
+	sendClusterMetadata(sender, clusterMessage, o.clusterID)
+}
+
+func sendNodesMetadata(sender aggregator.Sender, nodesList []*v1.Node, nodesMessages []model.MessageBody, clusterID string) {
 	stats := orchestrator.CheckStats{
-		CacheHits: len(nodesList) - len(messages),
-		CacheMiss: len(messages),
+		CacheHits: len(nodesList) - len(nodesMessages),
+		CacheMiss: len(nodesMessages),
 		NodeType:  orchestrator.K8sNode,
 	}
 
 	orchestrator.KubernetesResourceCache.Set(orchestrator.BuildStatsKey(orchestrator.K8sNode), stats, orchestrator.NoExpiration)
 
-	sender.OrchestratorMetadata(messages, o.clusterID, forwarder.PayloadTypeNode)
+	sender.OrchestratorMetadata(nodesMessages, clusterID, forwarder.PayloadTypeNode)
+}
+
+func sendClusterMetadata(sender aggregator.Sender, clusterMessage model.MessageBody, clusterID string) {
+	var stats orchestrator.CheckStats
+	if clusterMessage != nil {
+		stats = orchestrator.CheckStats{
+			CacheHits: 0,
+			CacheMiss: 1,
+			NodeType:  orchestrator.K8sCluster,
+		}
+		sender.OrchestratorMetadata([]serializer.ProcessMessageBody{clusterMessage}, clusterID, forwarder.PayloadTypeCluster)
+	} else {
```

Comment:
just double checking here that there's no confusion between the error case and the case hit case?
as I understood it if there's an error in `extractClusterMessage` the cluster message will be `nil` but we still count it as a cache hit

Reply:
Ah, you are right 💅 . Let me rework it!